src/vmm/src/vstate/vcpu/aarch64.rs

```diff
@@ -238,7 +238,7 @@ impl KvmVcpu {
 pub struct VcpuState {
     pub mp_state: kvm_bindings::kvm_mp_state,
     #[version(end = 2, default_fn = "default_old_regs")]
-    old_regs: Vec<Aarch64RegisterOld>,
+    pub old_regs: Vec<Aarch64RegisterOld>,
```

Comment:
Actually opposite. I prefer to have fields to be public. The prevention of changes to these fields should be with usage of `const` or `mut` modifiers.

Reply:
It's not about preventing modification of the field, it's about being able to change the implementation without breaking the anybody who consumes the API, e.g. if we ever change `old_regs` to be stored in different kind of container you can still have a getter that returns a `Vec<Aarch64RegisterOld>` so that anybody calling that function won't break.

Reply:
Considering we are the only ones consuming our internal API, we can handle the changes. Getters/setters just increase indirection and complexity and even in our code we don't use them much.

Reply:
Everybody can handle the changes. I'm just suggesting that we do it with the minimum amount of effort. While at the same time avoiding making public things that otherwise shouldn't be.

Reply:
I think in Rust, generally direct field access is preferred over getters/setters, mainly because Rust's ownership rules mean that there is multiple ways of "getting" data (e.g. a borrow vs moving), and also because direct field access allows for pattern matching, which is more idiomatic than a sequence of `if`s on `.get_(...)` calls.

Reply:
Ok, we're going into philosophical questions now :P AFAIK, for the time being, Rust seems to have issues with getting/moving struct fields comparing to the whole struct, but in any case, my main issue is about hiding implementation from the API, not ownership.

In any case, this is not a roadblock for me (marked as nit). Feel free to resolve Egor.